internal/resources/datasource_available_resources.go

```diff
@@ -407,19 +456,42 @@ func addVolmeFlavors(p *configuration.Config, d *schema.ResourceData, available
 	existingVols := make([]map[string]interface{}, 0, len(available.Volumes))
 	for _, vol := range available.Volumes {
 		iData := map[string]interface{}{
-			"id":         vol.ID,
-			vName:        vol.Name,
-			vDescription: vol.Description,
-			vSize:        vol.Capacity,
-			vLocationID:  vol.LocationID,
-			vFlavorID:    vol.FlavorID,
+			"id":           vol.ID,
+			vName:          vol.Name,
+			vDescription:   vol.Description,
+			vSize:          vol.Capacity,
+			vLocationID:    vol.LocationID,
+			vFlavorID:      vol.FlavorID,
+			vStoragePoolID: vol.StoragePoolID,
 		}
 		iData[sLocation], _ = p.GetLocationName(vol.LocationID)
 		iData[vFlavor], _ = p.GetVolumeFlavorName(vol.FlavorID)
+		iData[vStoragePool], _ = p.GetStoragePoolName(vol.StoragePoolID)
```

Comment:
specifying storage pool is not mandatory. If it is not given, we choose one. 
Here if storagePoolID is not given, this will fail, right? it should not.

Reply:
this is reading existing volumes which have storage pool information

Reply:
o ya, never mind then. create volume is taking care of if pool not given